### DISCUSSION TITLE
client_host null

### DISCUSSION THREAD

**Discussion by amervelic** [fastapi/fastapi#8886] [Questions] [answered]
```
from fastapi import FastAPI, Request

app = FastAPI()


@app.get("/items/{item_id}")
def read_root(item_id: str, request: Request):
    client_host = request.client.host
    return {"client_host": client_host, "item_id": item_id}

```

1. deploy to deta
2. host is null
3. https://204458.deta.dev/

**Comment by ricky-sb** [comment]
This appears to be an issue with Deta. Running this with `uvicorn` works just fine

**Comment by amervelic** [comment]
works on the heroku 💯  

```
import FastAPI,Request
app = FastAPI()

@app.get("/")
def getip(request:Request):
    return {"ip":request.headers['x-forwarded-for']}

```
https://fastip.herokuapp.com/

thanks !

**Comment by abdelhai** [comment, marked as answer]
Hi, Mustafa from Deta here. Due to a limitation in our global proxy, we are not forwarding the IP address at the moment. I hope this will be fixed soon. @amervelic
